token.go

```diff
@@ -391,6 +392,7 @@ func (t *BearerToken) getHeader(name string) (string, error) {
 // There's a deeper explanation of this in the Vault source code.
 // https://github.com/hashicorp/vault/blob/b17e3256dde937a6248c9a2fa56206aac93d07de/builtin/credential/aws/path_login.go#L1569
 func buildHttpRequest(method, endpoint string, parsedUrl *url.URL, body string, headers http.Header) (*http.Request, error) {
+	endpoint = strings.TrimSuffix(endpoint, "/")
```

Comment:
Instead of doubling down on string manipulation here, I think we should use a proper URL constructor

Before I found out you guys were this far along, I put together a test to demonstrate this issue
![image](https://github.com/user-attachments/assets/e05196f2-a6d1-4325-b783-07b222692ba1)

And a patch that fixes that issue
![image](https://github.com/user-attachments/assets/b6d51db9-22d2-48ef-8977-e07c2fc2a7b5)